lib/gemstash/gem_pusher.rb

```diff
@@ -38,12 +39,27 @@ def storage
       @storage ||= Gemstash::Storage.for("private").for("gems")
     end
 
+    def spec_storage
+      @spec_storage ||= Gemstash::Storage.for("private").for("specs")
+    end
+
+    def full_name
+      @full_name ||= gem.spec.full_name
+    end
+
     def check_auth
       Gemstash::Authorization.check(@auth_key, "push")
     end
 
     def store_gem
-      storage.resource(gem.spec.full_name).save(@content, indexed: true)
+      storage.resource(full_name).save(@content, indexed: true)
+    end
+
+    def store_gemspec
+      spec = gem.spec
+      spec = Marshal.dump(spec)
+      spec = Zlib::Deflate.deflate(spec)
+      spec_storage.resource(full_name).save(spec)
     end
 
```

Comment:
Heh, are we reaching the same concerns in 2 different branches? => https://github.com/bundler/gemstash/blob/pull-all-the-gems/lib/gemstash/preload.rb#L64-L85

Should we extract something that knows how to read/write gemspecs?